miner/worker.go

```diff
@@ -919,50 +921,64 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 
 	w.updateSnapshot()
 
-	// Play our part in generating the random beacon.
-	if w.isRunning() && random.IsRunning() {
-		istanbul, ok := w.engine.(consensus.Istanbul)
-		if !ok {
-			log.Crit("Istanbul consensus engine must be in use for the randomness beacon")
+	if w.isRunning() {
+		// TODO use caller instead of parentCaller (HF required)
+		randomIsActive := false
+		header := w.chain.CurrentHeader()
+		state, err := w.chain.StateAt(header.Root)
+		// TODO(HF) make this check use w.current.header instead of blockchain's current
+		if err == nil {
+			parentVMRunner := w.chain.NewEVMRunner(parent.Header(), state)
```

Comment:
This looks strange to me.  The origin `random.IsRunning()` uses the current chain head, whatever it may be.  Here you are using the state from the current chain head with the header from `parent`.  Now, if `parent` and `header` are from the same block, why not use `header` here instead?  On the other hand, if they're from different blocks, then the runner gets a header and a state which don't correspond.

My preference would be to just use a runner like `vmRunner` below which uses `w.current`, as discussed in the other comments.  But if instead you'd rather keep the old behavior, then this should use `header` instead of `parent.Header()` to match it more closely and be consistent in and of itself.

Reply:
Changed to use vmRunner from current header & state